### ISSUE TITLE
Depth Estimation for Moving objects to infinity Distance

### ISSUE THREAD

**Issue by forwardcjj** [TRI-ML/packnet-sfm#132]
Hi, @VitorGuizilini-TRI 
Thanks for your great work. 

Trying to finetune PackNet01_MR_velsup_CStoK with my own datasets. 

But got some weird depth map during the training step. 

I follow most of the overfitting kitti parameters and did not make changes. 

![Screen Shot 2021-03-31 at 6 31 10 PM](https://user-images.githubusercontent.com/1979795/113231240-a730d880-924f-11eb-967d-fc228b9f2d82.png)

**Comment by VitorGuizilini-TRI** [comment]
Unfortunately that's to be expected in a self-supervised model. When you have vehicles moving at the same speed as the camera there is no relative motion, so the network learns that these objects are at infinite depth (i.e. very far away, so parallax is zero). 

The monodepth2 paper proposes auto-masking to alleviate this issue, which we use in this repository, but apparently that's not enough. We propose an alternative solution in our ICLR'20 paper (https://arxiv.org/abs/2002.12319), feel free to check that out and let me know if you have any questions!

**Comment by forwardcjj** [comment]
@VitorGuizilini-TRI Great thanks for your quick reply.  Let me check the auto-masking part first.

**Comment by forwardcjj** [comment]
@VitorGuizilini-TRI I studied the auto-masking today and check the code in monodepth2. Still not clear 
identity_reprojection_loss += torch.randn(
                    identity_reprojection_loss.shape).cuda() * 0.00001

Do we have the same configuration in the current repo?  Thanks. 

https://github.com/nianticlabs/monodepth2/blob/ab2a1bf7d45ae53b1ae5a858f4451199f9c213b3/trainer.py#L468

 https://github.com/TRI-ML/packnet-sfm/blob/2698f1fb27785275ef847f3dbbd550cf8fff1799/packnet_sfm/losses/multiview_photometric_loss.py#L329

**Comment by designerZhou** [comment]
@forwardcjj @VitorGuizilini-TRI  I also encounter this problem ,do you solve this problem?

**Comment by UditSinghParihar** [comment]
I am also facing the same issue of infinite depth with moving objects. Would training in supervised way using depth maps or Lidar along with self-supervision helps?
